Add nt1m's <input type=checkbox switch> demo to webkit.org

annevk opened the pull request:
#### 1c73931b16512d904995e20c4e40db2c92bd559e
<pre>
Add nt1m&apos;s &lt;input type=checkbox switch&gt; demo to webkit.org
<a href="https://bugs.webkit.org/show_bug.cgi?id=268286">https://bugs.webkit.org/show_bug.cgi?id=268286</a>

Reviewed by Tim Nguyen.

A copy from <a href="https://github.com/nt1m/html-switch-demos">https://github.com/nt1m/html-switch-demos</a> but with adjusted
syntax for the content properties to make them not relevant for AT as
otherwise AT end users would end up with a worse experience.

* Websites/webkit.org/demos/html-switch/demo.css: Added.
* Websites/webkit.org/demos/html-switch/index.html: Added.

Canonical link: <a href="https://commits.webkit.org/274224@main">https://commits.webkit.org/274224@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ceb32c1c06a58a3dca7bb1b91d148a1788bf58c2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38291 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17236 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40610 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40849 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34062 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19996 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14574 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32303 "Found 1 new test failure: media/track/media-element-enqueue-event-crash.html (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38864 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14550 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12666 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12647 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42127 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34855 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34650 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38487 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13228 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10943 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36684 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14783 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8623 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13638 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14246 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.basic-with-symbol::before {
+    content: "〇";
```
It's probably worth changing this too `content: "〇" / ""` making use of WebKit's new support for the content alt syntax, this improves the experience for screen reader users.

nt1m replied:
It hasn't landed in STP despite what we claimed in the last release notes, but yeah I agree otherwise

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.basic-with-symbol:checked::before {
+    content: "I";
```
ditto here

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.sliding-track::before {
+    content: "ON";
```
ditto

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.sliding-track::after {
+    content: "OFF";
```
ditto

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.toggle-button::after {
+    content: "ON";
```
ditto

lukewarlow commented on Websites/webkit.org/demos/html-switch/index.html:
```diff
@@ -0,0 +1,359 @@
+}
+
+.toggle-button::before {
+    content: "OFF";
```
ditto

annevk committed:
Add nt1m's <input type=checkbox switch> demo to webkit.org
https://bugs.webkit.org/show_bug.cgi?id=268286

Reviewed by Tim Nguyen.

A copy from https://github.com/nt1m/html-switch-demos but with adjusted
syntax for the content properties to make them not relevant for AT as
otherwise AT end users would end up with a worse experience.

* Websites/webkit.org/demos/html-switch/demo.css: Added.
* Websites/webkit.org/demos/html-switch/index.html: Added.

Canonical link: https://commits.webkit.org/274224@main
